### ISSUE TITLE
PostgresUtils.toPGXYZString() methods (and others) should avoid using String.replace() pre JDK 9

### ISSUE THREAD

**Issue by ldaley** [jOOQ/jOOQ#7179]
The code in question is here: https://github.com/jOOQ/jOOQ/blob/master/jOOQ/src/main/java/org/jooq/util/postgres/PostgresUtils.java#L448

`String.replace()` looks like this:

```
    public String replace(CharSequence target, CharSequence replacement) {
        return Pattern.compile(target.toString(), Pattern.LITERAL).matcher(
                this).replaceAll(Matcher.quoteReplacement(replacement.toString()));
    }
```

So this method compiles the target regexes repeatedly, and generating new `Pattern` objects each time. This showed up as an allocation hotspot in code that I was profiling WRT object count (the size is quite low).

It might be worth compiling the patterns as constants and reusing.

**Comment by lukaseder** [comment]
Thanks for pointing this out. Interesting, this usage must've slipped by me when fixing #6672. JDK 9 finally implements `String.replace()` optimally, by not resorting to a wasteful `Pattern` internally. Until then, we better use our own (i.e. Apache Commons) `StringUtils`

**Comment by lukaseder** [comment]
Hmm, those `PostgresUtils.toXYZString()` methods would probably profit from additional refactoring where we pass around a `StringBuilder` as an "in out" parameter, rather than returning a new string all the time. A lot of intermediate `StringBuilder` allocations could be avoided.

Will profile this on my side.

**Comment by lukaseder** [comment]
I've created #7180 for the additional improvement.

**Comment by ldaley** [comment]
Sounds great. Thanks.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.11.0-SNAPSHOT. Will merge to 3.9.7 (#7182) and 3.10.6 (#7181)
